docs/developer/data_address_types.md

```diff
@@ -0,0 +1,124 @@
+
+##### Content Data Address (CDA)
+
+A Content Data Address (CDA) is created from an ADA. In pull scenarios, the CDA is sent to the consumer. in push scenarios, the provider uses the CDA to resolve asset content and send it to the consumer.
+
+An CDA can be created from an ADA by copying it and sending the latter to the consumer. The CDA can also be a derivative of the ADA. For example, the the CDA could reference data that was processed by a backend system provisioner. In this case, the backend provisioner would use the ADA to obtain the "raw" asset content, process it, and return a provision result containing an CDA pointing to the processed output.
+
+
+## Data Plane
+
+There are two types of data addresses in the data plane: Source Data Addresses and Destination Data Addresses. A source data address is where data originates; a destination data address is where data is transferred to. **Note that the source and destination addresses do not necessarily correspond to the consumer and provider respectively in a data transfer.** The following section enumerates the different cases.
+
+#### Push HTTP with Consumer and Provider DPFs
+
+A consumer requests data that is pushed by the provider over HTTP.
+
+###### Provider Data Plane (Originator)
+
+| Control Plane | DPF                 |
+|:--------------|:--------------------|
+| CDA           | Source Address      |
+| DDA           | Destination Address |
+
+###### Consumer Data Plane
+
+| Control Plane  | DPF                |
+|:---------------|:-------------------|
+| DDA            | Source Address     |
+| Consumer Sink  | Destination Address|
+
+
+#### Pull HTTP with Consumer and Provider DPFs
+
+A consumer requests data that it pulls from the provider over HTTP.
+
+
+###### Consumer Data Plane (Originator)
+
+| Control Plane  | DPF                |
+|:---------------|:-------------------|
+| CDA            | Source Address     |
+| Consumer Sink  | Destination Address|
+
+
+###### Provider Data Plane
+
+| Control Plane  | DPF                 |
+|:---------------|:--------------------|
+| CDA            | Source Address      |
+| Provider Sink  | Destination Address |
+
+
+#### Push Object Storage
+
+A consumer requests data that is pushed by the provider to the consumer's object storage.
+
+###### Provider Data Plane (Originator) 
+
+| Control Plane  | DPF                 |
+|:---------------|:--------------------|
+| CDA            | Source Address      |
+| DDA            | Destination Address |
+
+###### Consumer Data Plane
+
+No consumer DPF is engaged.
+
+#### Pull Object Storage
+
+A consumer requests data that it pull from the provider's object storage.
+
+###### Consumer Data Plane (Originator)
+
+| Control Plane  | DPF                 |
+|:---------------|:--------------------|
+| CDA            | Source Address      |
+| Consumer Sink  | Destination Address |
```

Comment:
Who triggers the actual data transfer here?
My understanding that only the provider executes the `DataFlowController`, but maybe I am wrong. 
If I am correct, then how consumer can be the originator of the data transfer?